include/deal.II/base/tensor.h

```diff
@@ -826,7 +826,7 @@ template <int dim, typename Number>
 inline DEAL_II_CUDA_HOST_DEV Tensor<0, dim, Number>::operator Number &()
 {
   // We cannot use Assert inside a CUDA kernel
-#ifndef __CUDA_ARCH__
```

Comment:
The only reason for relaxing this check is that we don't need to use both `__CUDA_ARCH__` and `__CUDACC__`.
It should be fine to also compile the `host` code using `nvcc` with the `Assert`enabled. Do you have a strong opinion?

Reply:
Not really. I did that to use `__CUDACC__` everywhere after that I had some problem using `__CUDA_ARCH__` somewhere else. I'll change it back